types/src/v2/response.rs

```diff
@@ -1,30 +1,45 @@
-use crate::v2::params::{JsonRpcNotificationParamsAlloc, TwoPointZero};
+use crate::v2::params::{Id, JsonRpcNotificationParams, JsonRpcNotificationParamsAlloc, TwoPointZero};
 use serde::{Deserialize, Serialize};
-use serde_json::value::RawValue;
 
 /// JSON-RPC successful response object.
 #[derive(Serialize, Deserialize, Debug)]
+#[serde(deny_unknown_fields)]
 pub struct JsonRpcResponse<'a, T> {
 	/// JSON-RPC version.
 	pub jsonrpc: TwoPointZero,
 	/// Result.
 	pub result: T,
 	/// Request ID
 	#[serde(borrow)]
-	pub id: Option<&'a RawValue>,
+	pub id: Id<'a>,
+}
+
+/// JSON-RPC subscription response.
+#[derive(Serialize)]
+pub struct JsonRpcSubscriptionResponse<'a> {
+	/// JSON-RPC version.
+	pub jsonrpc: TwoPointZero,
+	/// Method
+	pub method: &'a str,
+	/// Params.
+	pub params: JsonRpcNotificationParams<'a>,
```

Comment:
In another PR; merge `JsonRpcNotificationParams` and `JsonRpcNotificationParamsNotification`